ocs_ci/helpers/dr_helpers_ui.py

```diff
@@ -0,0 +1,274 @@
+        wait (bool): When True, cluster status check will be done for the expected text/status in the given timeout.
+
+    """
+
+    ocp_version = version.get_semantic_ocp_version_from_config()
+    acm_obj = AcmPageNavigator(setup_acm_ui)
+    acm_loc = locators[ocp_version]["acm_page"]
+
+    acm_obj.navigate_clusters_page()
+
+    if down_cluster_name:
+        wait = True
+        check_cluster_unavailability = acm_obj.wait_until_expected_text_is_found(
+            format_locator(acm_loc["cluster_status_check"], down_cluster_name),
+            expected_text="Unavailable",
+            timeout=timeout,
+        )
+        if check_cluster_unavailability:
+            log.info(f"Down cluster {down_cluster_name} is {expected_text}")
+            return
+        else:
+            cluster_status = acm_obj.get_element_text(acm_loc["cluster_status_check"])
+            assert (
+                cluster_status == "Ready"
+            ), f"Down cluster {down_cluster_name} is still in {cluster_status} state after {timeout} seconds"
+            if not check_cluster_unavailability and cluster_status != "Ready":
+                other_expected_status = ["NotReady", "Error", "Unknown"]
+                for status in other_expected_status:
+                    check_cluster_unavailability = (
+                        acm_obj.wait_until_expected_text_is_found(
+                            format_locator(
+                                acm_loc["cluster_status_check"], down_cluster_name
+                            ),
+                            expected_text=status,
+                            timeout=30,
+                        )
+                    )
+                    if check_cluster_unavailability:
+                        f"Cluster {down_cluster_name} is in {status} state"
+                        return
+                return
+            else:
+                log.error(
+                    f"Down cluster {down_cluster_name} status check failed, actual status is {cluster_status}"
+                )
+                raise NoSuchElementException
+    if not cluster_names:
+        primary_cluster = get_current_primary_cluster_name
+        secondary_cluster = get_current_secondary_cluster_name
+        cluster_names = ["local-cluster", primary_cluster, secondary_cluster]
+    for cluster in cluster_names:
+        acm_obj.do_click(format_locator(acm_loc["cluster_name"], cluster))
+        cluster_status = acm_obj.get_element_text(acm_loc["cluster_status_check"])
+        log.info(f"Cluster {cluster} status is {cluster_status} on ACM UI")
+        if wait:
+            wait_cluster_readiness = acm_obj.wait_until_expected_text_is_found(
+                format_locator(acm_loc["cluster_status_check"], cluster),
+                expected_text=expected_text,
+                timeout=timeout,
+            )
+            assert wait_cluster_readiness
```

Comment:
better if the assertion exception is raised in the test scope and not the util function. From the util function we should return a value which would be later examined in the test

Reply:
Done